### PR TITLE
Hide navigation items instead of simply not rendering them and updated dependencies

### PR DESCRIPTION
**Pull request by sroy3**

### CODE REVIEW

**Comment by MichelML** [comment]
So I don't approve blindly, why are we making this change exactly?

**Comment by sroy3** [comment]
If the pagination is not rendered, the pagination state won't be initialized. Before parsing the state from the url in the admin we are waiting on some element state to be initialized before changing them. If the pagination state does not exist at first, we will not be able to change the state to the one saved in the url.

File: src/components/navigation/Navigation.tsx [hunk 1]
```diff
@@ -34,48 +34,50 @@ export interface INavigationStateProps extends IReduxStatePossibleProps {
...
+    if (this.props.currentPerPage) {
+      perPageProps.currentPerPage = this.props.currentPerPage;
     }
+    const perPage: JSX.Element = this.props.withReduxState
+      ? <NavigationPerPageConnected id={this.props.id} loadingIds={this.props.loadingIds} {...perPageProps} />
+      : <NavigationPerPage onPerPageClick={this.props.onPerPageClick} {...perPageProps} />;
+    const perPageClass = this.props.perPageNumbers.length && this.props.totalEntries > this.props.perPageNumbers[0] ? '' : 'hidden';
 
     let navigationClasses: string = 'pagination-container' + (this.props.isLoading ? ' loading-view' : '');
     let loading: JSX.Element = this.props.withReduxState ? <LoadingConnected id={'loading-' + this.props.id} /> : <Loading />;
```
Comment (MichelML) [inline src/components/navigation/Navigation.tsx]: i know this is out of scope but you could use const and template strings here

``` 
const navigationClasses: string = `pagination-container${this.props.isLoading ? ' loading-view' : ''}`;
const loading: JSX.Element = this.props.withReduxState ? <LoadingConnected id={`loading-${this.props.id}`} /> : <Loading />;

File: src/components/navigation/Navigation.tsx [hunk 2]
```diff
@@ -34,48 +34,50 @@ export interface INavigationStateProps extends IReduxStatePossibleProps {
...
-    }
+    const paginationProps: INavigationPaginationProps = {
+      totalPages: this.props.totalPages,
+      numberOfPagesToShow: this.props.numberOfPagesToShow,
+      previousLabel: this.props.previousLabel,
+      nextLabel: this.props.nextLabel,
+      hidePages: this.props.hidePages || !this.props.totalPages || this.props.totalPages === 1
+    };
+    const pagination: JSX.Element = this.props.withReduxState
+      ? <NavigationPaginationConnected id={'pagination-' + this.props.id} loadingIds={this.props.loadingIds} {...paginationProps} />
```
Comment (MichelML) [inline src/components/navigation/Navigation.tsx]:
```
id={`pagintation-${this.props.id}`}
```

**Pull request merged**